WordPress/Classes/ViewRelated/Reader/ReaderMenuViewController.swift

```diff
@@ -87,6 +120,75 @@ import WordPressShared
     // MARK: - Instance Methods
 
 
+    /// When logged out return the nav stack to the menu
+    ///
+    func handleAccountChanged(notification: NSNotification) {
+        // Return to the root vc.
+        navigationController?.popToRootViewControllerAnimated(false)
+
+        // Clear the flag so Discover will be present and ready to view.
+        NSUserDefaults.standardUserDefaults().removeObjectForKey(readerHasBeenPreviouslyViewedKey)
+
+        // Clean up obsolete content.
+        let context = ContextManager.sharedInstance().mainContext
+        ReaderTopicService(managedObjectContext: context).deleteAllTopics()
+        ReaderPostService(managedObjectContext: context).deletePostsWithNoTopic()
+
+        // Sync the menu fresh
+        syncTopics()
+    }
+
+
+    /// The first time the Reader is launched, we want to show the Discover topic,
+    /// not the menu.
+    ///
+    func handleFirstLaunchIfNeeded() {
+        let defaults = NSUserDefaults.standardUserDefaults()
+        if defaults.boolForKey(readerHasBeenPreviouslyViewedKey) == true {
```

Comment:
It might be a bit more clear and maintainable to wrap the use of the `readerHasBeenPreviouslyViewedKey` in helper methods such as `readerHasBeenPreviouslyViewed` and `setReaderHasBeenPreviouslyViewed:`.